### ISSUE TITLE
Usage of `jq` blobs?

### ISSUE THREAD

**Issue by aegershman** [18F/cg-nessus-manager-boshrelease#27]
**tl;dr** how are you adding `jq/jq-1.5` as a blob? Do you already expect it to be your director's blobstores?

`jq` is a requirement of the nessus-manager job + is listed [as part of the blob config](https://github.com/18F/cg-nessus-manager-boshrelease/blob/master/config/blobs.yml#L1-L4), but how is the `jq` blob getting added? Is it downloaded from [github as `jq-linux64`](https://github.com/stedolan/jq/releases/tag/jq-1.5), gzipped as `jq-1.5.gz`, and then added as a blob?

Is there a reason it isn't in the `src/` folder in `git`?

For example, on bosh-lite, after downloading the `jq` tarball & renaming it, here's what I did to get `jq` in my local blobstore:

```sh
    bosh -e vbox add-blob \
        --dir=/Users/aeg/projects/cg-nessus-manager-boshrelease \
        ~/projects/cg-nessus-manager-boshrelease/src/jq/jq-1.5.gz \
        'jq/jq-1.5.gz'
```

Is this accurate? Advice welcome, thank you in advance.

**Comment by bengerman13** [comment]
hi @aegershman 
the blob in question is a single binary, like [this one](https://github.com/stedolan/jq/releases/download/jq-1.5/jq-linux64), but gzipped and renamed.

i.e.
```
wget https://github.com/stedolan/jq/releases/download/jq-1.5/jq-linux64
mv jq-linux64 jq-1.5
gzip jq-1.5
bosh add-blob jq-1.5.gz jq/jq-1.5.gz
```

It's not under the `src` directory because that's intended for code maintained within the boshrelease, not for vendoring external dependencies.

I'm going to close this so it doesn't go stale if this answers your question, but feel free to reply if you have any other questions!

**Comment by aegershman** [comment]
Thank you for your response; could I ask how you all get the `jq` blob (and `nessus`) into your bosh directors using automation, then? Is that handled in pipelines somewhere?

I'm still working out the best way to automate the collection/uploading of all required external dependencies (in this case just nessus and jq) so a release can be created & uploaded to other directors.

Maybe it's in a private repo, but I couldn't tell where these blobs are coming from on the ci [in this repo](https://github.com/18F/cg-nessus-manager-boshrelease/blob/master/ci/pipeline.yml) and in the [deploy-nessus-manager](https://github.com/18F/cg-deploy-nessus-manager/blob/master/pipeline.yml) pipeline + the corresponding tasks... Any hints are appreciated 😉

Apologies, don't mean to be overbearing with questions, just trying understand how you all are doing it 👍

**Comment by bengerman13** [comment]
we upload these blobs manually - to consume this release you'll have to maintain your own `blobs.yml`, since bosh doesn't have a way to force a blob's ID. The way we've tended to do this when we consume upstream releases is by maintaining a fork.

The process looks something like:
1. locate and download blobs
2. add them using `bosh add-blobs`
3. upload them using `bosh upload-blobs`
4. commit and push `config/blobs.yml` to your fork

**Comment by aegershman** [comment]
Fair enough, this makes sense. Just a few other things, thanks again for your responses--

The [blobstore 18F has listed for these boshreleases](https://github.com/18F/cg-nessus-manager-boshrelease/blob/master/config/final.yml#L6) isn't publicly exposed for someone to download from or `bosh sync-blobs` against. It also isn't packaged into a `nessus-manager-release.tgz` and uploaded as a github release. Is this because of legal reasons around redistribution? Or maybe it just not being as much of a priority to automate it as being a publicly-consumable bosh release?

... I promise I don't mean that to sound accusatory or salty, I'm just trying to get bearings before our team starts rolling out `nessus` on our bosh deployments internally.

thanks 👍

**Comment by bengerman13** [comment]
for this release specifically, we aren't authorized to redistribute the closed-source binaries from Nessus.
More generally for the rest of our releases - it's mostly about managing complexity. If we could expose every blob, it wouldn't be that difficult to have a public blobstore, but if we have even a single non-public blob, managing what blobs are in what store gets very complicated, since bosh only lets you consume from one blobstore in any single release

**Comment by aegershman** [comment]
@bengerman13 I've learned a lot from this, thank you. We often consume other OSS bosh releases directly and not had to think much about _creating_ the release; i.e. being responsible for gathering the blobs from both OSS and closed-source, storing them, cutting the final release, etc.

In general, are you all okay with me of these PRs I'm opening against the nessus-manager & nessus-agent? Don't mean to impose expectation of any of it getting merged or prioritized. If none of them get merged, or you all have no intention of changing any of this, no worries at all I'll back off on these PRs 👌

**Comment by bengerman13** [comment]
PRs welcome 👍
